lib/matplotlib/tests/test_datetime.py

```diff
@@ -612,11 +612,28 @@ def test_violinplot(self):
         fig, ax = plt.subplots()
         ax.violinplot(...)
 
-    @pytest.mark.xfail(reason="Test for vlines not written yet")
     @mpl.style.context("default")
     def test_vlines(self):
-        fig, ax = plt.subplots()
-        ax.vlines(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        fig, (ax1, ax2, ax3) = plt.subplots(3, 1, layout='constrained')
+        dates = [datetime.datetime(2023, 2, 10),
+                 datetime.datetime(2023, 5, 18),
+                 datetime.datetime(2023, 6, 6)]
+        ax1.set_xlim(left=datetime.datetime(2023, 1, 1),
+                     right=datetime.datetime(2023, 6, 30))
+        ax1.vlines(x=dates,
+                   ymin=[0, 0.25, 0.5],
+                   ymax=[0.25, 0.5, 0.75])
+        ax2.set_xlim(left=0,
+                     right=0.5)
+        ax2.vlines(x=[0.3, 0.35],
+                   ymin=[np.datetime64('2023-03-20'), np.datetime64('2023-03-31')],
+                   ymax=[np.datetime64('2023-05-01'), np.datetime64('2023-05-16')])
+        ax3.set_xlim(left=datetime.datetime(2023, 7, 1),
+                     right=datetime.datetime(2023, 12, 31))
+        ax3.vlines(x=[datetime.datetime(2023, 9, 1), datetime.datetime(2023, 12, 10)],
```

Comment:
Any reason not to re-use `dates` on this one?

Reply:
I wanted to make each graph look fairly different from one other, so I tried to pick different dates and x/y limits. Initially, the `dates` array was too long to fit on one line which is why I made it a separate variable. If you'd prefer I re-use it, I can make the change but would have to modify the x limit as well because the values in `dates` fall outside of the current range for ax3. Let me know what you think

Reply:
Alternatively, I could get rid of `dates` altogether and move the array directly into `ax1.vlines()` but keep each element on a different line like it is right now

Reply:
Yea, I think it'd be clearer to not have `dates` if it's not being reused in other places with dates.

Reply:
Got it. Just fixed it